packages/migrate/src/commands/DbExecute.ts

```diff
@@ -100,6 +100,8 @@ ${chalk.bold('Examples')}
       return this.help(args.message)
     }
 
+    await checkUnsupportedDataProxy('db execute', args, !args['--url'])
```

Comment:
Only implicitly load the schema if `args['--url']` is not passed (see logic in the code).

Reply:
If I understand this correctly ("Only try to load a schema file if no explicit `--url` is given" - correct?), I think the same logic should apply to `db pull` then. It also has an (undocumented) param `--url` that can be used to override the connection string from the schema file.

Reply:
It looks like it is not the case anymore https://github.com/prisma/prisma/blob/5aadcaf52c3832ad52d7f4749edc698217aa8db5/packages/migrate/src/commands/DbPull.ts/#L167-L168, unless `.when` runs in reverse (but I doubt that). (Also arguably harder to read than what it was before). I might have overlooked something here...

Reply:
I do not even pretend that I can still read that code 🤷 

I go from my experience, which is that `--url` works on `db pull` both with no schema present (then you get a schema file with that URL as datasource and the introspection result afterwards) and with a schema present (then the datasource.url value in there is overwritten with what is provided to the CLI).

I think the command describes my understanding of the behavior: https://github.com/prisma/prisma/blob/5aadcaf52c3832ad52d7f4749edc698217aa8db5/packages/migrate/src/commands/DbPull.ts#L155-L162

Reply:
@jkomyno recently rewrote that code, so should be able to confirm.